seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/CoordinatorService.java

```diff
@@ -468,6 +468,13 @@ public PassiveCompletableFuture<Void> submitJob(long jobId, Data jobImmutableInf
         executorService.submit(
                 () -> {
                     try {
+                        if (!isStartWithSavePoint
+                                && getJobHistoryService().getJobMetrics(jobId) != null) {
+                            throw new JobException(
+                                    String.format(
+                                            "The job id %s has already been submitted and is not starting with a savepoint.",
+                                            jobId));
+                        }
```

Comment:
can we move this step to `ClientJobExecutionEnvironment` or `RestJobExecutionEnvironment`

![image](https://github.com/apache/seatunnel/assets/28497357/59978cb0-b0fd-4a60-ba7f-bd021e4bce87)

Reply:
forget this, it can‘t check the case that use same job id submit at the same time